staging/src/k8s.io/dynamic-resource-allocation/controller/controller.go

```diff
@@ -690,11 +742,18 @@ func (ctrl *controller) syncPodSchedulingContexts(ctx context.Context, schedulin
 				Name:     pod.Name,
 				UID:      pod.UID,
 			}
+
+			ctrl.allocateClaims(ctx, claims, selectedNode, selectedUser)
+
+			allocationFailed := false
 			for _, delayed := range claims {
-				if err := ctrl.allocateClaim(ctx, delayed.Claim, delayed.ClaimParameters, delayed.Class, delayed.ClassParameters, selectedNode, selectedUser); err != nil {
-					return fmt.Errorf("allocation of pod claim %s failed: %v", delayed.PodClaimName, err)
+				if delayed.Error != nil {
+					allocationFailed = true
 				}
 			}
+			if allocationFailed {
+				return fmt.Errorf("allocation of one or more pod claims failed")
```

Comment:
This should include the information provided by the driver. If there's exactly one error, return that. If there are many, combine them (somehow).